src/head.rs

```diff
@@ -63,12 +63,15 @@ struct PathIdx(usize);
 
 impl<T: Send + 'static + std::str::FromStr, S: 'static> Extract<S> for Path<T> {
     type Fut = future::Ready<Result<Self, Response>>;
-    fn extract(data: &mut S, req: &mut Request, params: &RouteMatch<'_>) -> Self::Fut {
+    fn extract(data: &mut S, req: &mut Request, params: &Option<RouteMatch<'_>>) -> Self::Fut {
         let &PathIdx(i) = req.extensions().get::<PathIdx>().unwrap_or(&PathIdx(0));
         req.extensions_mut().insert(PathIdx(i + 1));
-        match params.vec[i].parse() {
-            Ok(t) => future::ok(Path(t)),
-            Err(_) => future::err(http::status::StatusCode::BAD_REQUEST.into_response()),
+        match params {
+            Some(params) => match params.vec[i].parse() {
+                Ok(t) => future::ok(Path(t)),
+                Err(_) => future::err(http::status::StatusCode::BAD_REQUEST.into_response()),
+            },
+            None => future::err(http::status::StatusCode::BAD_REQUEST.into_response()),
```

Comment:
I think trying to extract match information from default handlers is a bug in application code. One should know that there's no wildcard match in default handler. Maybe `INTERNAL_SERVER_ERROR` is better, or... should panic here? 😝

Reply:
Haha yeah...I am hesitant to panic but I did make it `INTERNAL_SERVER_ERROR`. I guess I could make it panic? I dunno what to do here 😢